hi-ml/testhiml/testhiml/test_run_ml.py

```diff
@@ -48,6 +64,38 @@ def ml_runner_with_container() -> Generator:
         shutil.rmtree(output_dir)
 
 
+@pytest.fixture(scope="module")
```

Comment:
I think if you re-design your tests, you could save on some code. You could separate them out into two parts:
- First testing that all of the various ways of specifying a checkpoint give you a valid local file (effectively, testing `get_local_checkpoints_path_or_download`)
- Once you know that you can get a valid local file with all 3 options, you only need to test if the runner correctly picks up the local weights.